app/uk.gov.hmrc.helptosavestridefrontend/controllers/StrideController.scala

```diff
@@ -193,35 +192,48 @@ class StrideController @Inject() (val authConnector:       AuthConnector,
 
   def createAccount: Action[AnyContent] = authorisedFromStride { implicit request ⇒
     checkSession(SeeOther(routes.StrideController.getEligibilityPage().url),
-                 whenEligible   = { (eligible, detailsConfirmed, nsiUserInfo) ⇒
+                 whenEligible   = { (eligible, detailsConfirmed, nsiUserInfo, _) ⇒
         if (!detailsConfirmed) {
           SeeOther(routes.StrideController.customerEligible().url)
         } else {
-          helpToSaveConnector.createAccount(CreateAccountRequest(nsiUserInfo, eligible.response.reasonCode, "Stride")).fold(
+          helpToSaveConnector.createAccount(CreateAccountRequest(nsiUserInfo, eligible.response.reasonCode, "Stride")).fold[Future[Result]](
             error ⇒ {
               logger.warn(s"error during create account call, error: $error")
               SeeOther(routes.StrideController.getErrorPage().url)
             }, {
-              case AccountCreated ⇒
-                SeeOther(routes.StrideController.getAccountCreatedPage().url)
+              case AccountCreated(accountNumber) ⇒
+                sessionStore.store(HtsSession(eligible, nsiUserInfo, detailsConfirmed, Some(accountNumber))).fold(
+                  error ⇒ {
+                    logger.warn(s"error during updating session with accountNumber: $error")
+                    SeeOther(routes.StrideController.getErrorPage().url)
+                  },
+                  _ ⇒ SeeOther(routes.StrideController.getAccountCreatedPage().url)
+                )
+
               case AccountAlreadyExists ⇒
                 Ok(views.html.account_already_exists())
-            })
+            }).flatMap(identity)
```

Comment:
does `flatten` work here? You can avoid having to do `flatMap(identity)` by making use of `EitherT` here and flatMapping  from `createAccount` into something which will update the session if the account is created and then mapping into the result

Reply:
fixed